.github/workflows/assets/validate-registry/main.go

```diff
@@ -39,7 +39,16 @@ func main() {
 	}
 
 	if !reflect.DeepEqual(rawRepos, filteredRepos) {
-		fmt.Fprintln(os.Stderr, "error: Registry data file contains duplicate entries")
+		fmt.Fprintln(os.Stderr, "error: Registry data file contains duplicate URLs")
 		os.Exit(1)
 	}
+
+	nameMap := make(map[string]bool)
+	for index, entry := range rawRepos {
+		nameMap[entry.LibraryName] = true
+		if len(nameMap) < index+1 {
+			fmt.Fprintf(os.Stderr, "error: Registry data file contains duplicates of name %s\n", entry.LibraryName)
+			os.Exit(1)
+		}
```

Comment:
I think this could be made clearer like so, it should also be a bit more efficient since we don't check the size of the map each iteration.
```suggestion
	if _, found := nameMap[entry.LibraryName]; found {
		fmt.Fprintf(os.Stderr, "error: Registry data file contains duplicates of name %s\n", entry.LibraryName)
		os.Exit(1)
	}
	nameMap[entry.LibraryName] = true
```

Reply:
OK, done. Thanks!